server/routes/auth.js

```diff
@@ -1,20 +1,21 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
-const { validateRegister, validateLogin } = require('../validate');
-const protect = require('../middleware/auth');
-const {
-  registerUser,
-  loginUser,
-  loadUser,
-  logoutUser,
-} = require('../controllers/auth');
+const passport = require("passport");
+const { passportProtect } = require("../middleware/auth");
+const { loadUser, logoutUser, checkUserEmail } = require("../controllers/auth");
 
-router.route('/register').post(validateRegister, registerUser);
+router.route("/user").get(passportProtect, loadUser);
 
-router.route('/login').post(validateLogin, loginUser);
+router.route("/logout").get(logoutUser);
 
-router.route('/user').get(protect, loadUser);
+router.route("/email/:email").post(checkUserEmail);
 
-router.route('/logout').get(logoutUser);
+router
+  .route("/google")
+  .get(passport.authenticate("google", { scope: ["profile", "email", "https://www.googleapis.com/auth/calendar"] }));
+
+router.route("/google/callback").get(passport.authenticate("google", { failureRedirect: "/failure" }), (req, res) => {
```

Comment:
Does the failureRedirect route even exist?

Reply:
fix unused redirect 274758c